src/pycsou/opt/solver/pds.py

```diff
@@ -0,0 +1,965 @@
+
+    with :math:`\mathbf{D}\in\mathbb{R}^{N\times N}` the discrete derivative operator and :math:`\mathbf{G}\in\mathbb{R}^{L\times N}, \, \mathbf{y}\in\mathbb{R}^L, \lambda_1,\lambda_2>0.`
+    This problem can be solved via PDS with :math:`\mathcal{F}(\mathbf{x})= \frac{1}{2}\left\|\mathbf{y}-\mathbf{G}\mathbf{x}\right\|_2^2`, :math:`\mathcal{G}(\mathbf{x})=\lambda_2\|\mathbf{x}\|_1,`
+    :math:`\mathcal{H}(\mathbf{x})=\lambda \|\mathbf{x}\|_1` and :math:`\mathbf{K}=\mathbf{D}`.
+
+    .. plot::
+    >>> import matplotlib.pyplot as plt
+    >>> import numpy as np
+    >>> from pycsou.opt.solver.pds import PDS
+    >>> from pycsou._dev import FirstDerivative, DownSampling, SquaredL2Norm, L1Norm
+
+    >>> x = np.repeat(np.asarray([0, 2, 1, 3, 0, 2, 0]), 10)
+    >>> D = FirstDerivative(size=x.size, kind="forward")
+    >>> D.lipschitz(tol=1e-3)
+    >>> downsampling = DownSampling(size=x.size, downsampling_factor=3)
+    >>> downsampling.lipschitz()
+    >>> y = downsampling(x)
+    >>> l22_loss = (1 / 2) * SquaredL2Norm().as_loss(data=y)
+    >>> fidelity = l22_loss * downsampling
+    >>> H = 0.1 * L1Norm()
+
+    >>> G = 0.01 * L1Norm()
+    >>> pds = PDS(f=fidelity, g=G, h=H, K=D)
+    >>> x0, z0 = x * 0, x * 0
+    >>> pds.fit(x0=x0, z0=z0)
+
+    >>> estimate = pds.solution()
+    >>> x_recons = estimate[0]
+    >>>
+    >>> plt.figure()
+    >>> plt.stem(x, linefmt="C0-", markerfmt="C0o")
+    >>> mask_ids = np.where(downsampling.downsampling_mask)[0]
+    >>> markerline, stemlines, baseline = plt.stem(mask_ids, y, linefmt="C3-", markerfmt="C3o")
+    >>> markerline.set_markerfacecolor("none")
+    >>> plt.stem(x_recons, linefmt="C1--", markerfmt="C1s")
+    >>> plt.legend(["Ground truth", "Observation", "PDS Estimate"])
+    >>> plt.show()
+
+    See Also
+    --------
+    :py:class:`~pycsou.opt.solver.pds.PDS`, :py:class:`~pycsou.opt.solver.pds.ChambollePockSplitting`, :py:class:`~pycsou.opt.solver.pds.DouglasRachford`
+    """
+
+    def __init__(
+        self,
+        f: typ.Optional[pyco.DiffFunc] = None,
+        g: typ.Optional[pyco.ProxFunc] = None,
+        h: typ.Optional[pyco.ProxFunc] = None,
+        K: typ.Optional[pyco.LinOp] = None,
+        *,
+        folder: typ.Optional[pyct.PathLike] = None,
+        exist_ok: bool = False,
+        writeback_rate: typ.Optional[int] = None,
+        verbosity: int = 1,
+        log_var: pyct.VarName = ("x",),
+    ):
+        super().__init__(
+            folder=folder,
+            exist_ok=exist_ok,
+            writeback_rate=writeback_rate,
+            verbosity=verbosity,
+            log_var=log_var,
+        )
+
+        self._f = pyclo.NullFunc() if (f is None) else f
+        self._g = pyclo.NullFunc() if (g is None) else g
+        self._h = pyclo.NullFunc() if (h is None) else h
+        if h is not None:
+            if K is None:
+                self._K = pyclo.IdentityOp()
+                self._K._lipschitz = 1
```

Comment:
No need the Lipschitz constant of the ``IdentityOp`` is set to one at construction time.